### PR TITLE
Add filter to count pixel tracks for beta star 90m run

### PR DESCRIPTION
**Pull request by JanFSchulte**
Adds a filter to count pixel tracks at HLT needed for high beta star run.

### CODE REVIEW

**Comment by fwyzard** [comment]
@JanFSchulte do you need the `HLTFilter`-specific functionality of saving the candidates ?
Otherwise it would simpler to write an `edm::global::EDFilter`.

File: HLTrigger/special/plugins/HLTPixelTrackFilter.cc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  // get hold of products from Event
+  edm::Handle< reco::TrackCollection> trackColl;
+  event.getByToken(inputToken_, trackColl);
+
+  unsigned int numTracks = trackColl->size();
+  LogDebug("") << "Number of tracks accepted: " << numTracks;
+  bool accept = (numTracks >= min_pixelTracks_);
+  if(max_pixelTracks_ > 0)
+    accept &= (numTracks <= max_pixelTracks_);
+  std::cout << numTracks << std::endl;
```
Comment (fwyzard) [inline HLTrigger/special/plugins/HLTPixelTrackFilter.cc]: No `cout`, please use a `LogDeug` or `LogInfo`.

**Comment by Martin-Grunewald** [comment]
This filter does not use the functionality of an HLTFilter, so should be a normal EDFilter.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@JanFSchulte 
We would need a 101X backport PR!

**Comment by JanFSchulte** [comment]
@Martin-Grunewald See #23485 for the backport.

**Comment by fabiocos** [comment]
@JanFSchulte @Martin-Grunewald where is this filter effectively tested?

**Comment by JanFSchulte** [comment]
@fabiocos studies using this filter have been shown at the TSG last week https://indico.cern.ch/event/733179/contributions/3024006/attachments/1658757/2656741/CMSTOTEM_TSGmeeting_30May2018_Robert.pdf slide 6ff

The layer counting works as expected and the thresholds are being tuned.

**Comment by fabiocos** [comment]
+1

**Pull request merged**